core/modules/widgets/reveal.js

```diff
@@ -34,23 +34,27 @@ RevealWidget.prototype.render = function(parent,nextSibling) {
 	if(this.revealTag && $tw.config.htmlUnsafeElements.indexOf(this.revealTag) === -1) {
 		tag = this.revealTag;
 	}
-	var domNode = this.document.createElement(tag);
-	var classes = this["class"].split(" ") || [];
-	classes.push("tc-reveal");
-	domNode.className = classes.join(" ");
-	if(this.style) {
-		domNode.setAttribute("style",this.style);
-	}
-	parent.insertBefore(domNode,nextSibling);
-	this.renderChildren(domNode,null);
-	if(!domNode.isTiddlyWikiFakeDom && this.type === "popup" && this.isOpen) {
-		this.positionPopup(domNode);
-		$tw.utils.addClass(domNode,"tc-popup"); // Make sure that clicks don't dismiss popups within the revealed content
-	}
-	if(!this.isOpen) {
-		domNode.setAttribute("hidden","true");
+	if(this.revealTag || this.class || this.style || this.type === "popup" || this.animate === "yes" || this.retain === "yes"){
+		var domNode = this.document.createElement(tag);
+		var classes = this["class"].split(" ") || [];
+		classes.push("tc-reveal");
+		domNode.className = classes.join(" ");
+		if(this.style) {
+			domNode.setAttribute("style",this.style);
+		}
+		parent.insertBefore(domNode,nextSibling);
+		this.renderChildren(domNode,null);
+		if(!domNode.isTiddlyWikiFakeDom && this.type === "popup" && this.isOpen) {
+			this.positionPopup(domNode);
+			$tw.utils.addClass(domNode,"tc-popup"); // Make sure that clicks don't dismiss popups within the revealed content
+		}
+		if(!this.isOpen) {
+			domNode.setAttribute("hidden","true");
+		}
+		this.domNodes.push(domNode);
+	} else {
+		this.renderChildren(parent, nextSibling);
```

Comment:
hmmm, I'm not 100% sure, but if `retain` is `no`, and reveal is closed, nothing should be rendered. right?

Reply:
The current behavior is that the node is still render with the hidden attribute set and no content.